src/app/model/Rock.ts

```diff
@@ -0,0 +1,54 @@
+export class Rock implements PieceMovement {
+  id: string;
+  side: string;
+  position: string;
+
+  constructor(id: string, side: string, position: string) {
+    this.id = id;
+    this.side = side;
+    this.position = position;
+  }
+  possibleMoves(position: string, board: Board): string[] {
```

Comment:
To jest bardzo spoko, że position przekazujesz tutaj. Tylko w takim razie nie jest nam potrzebne jako pole klasy :)

Reply:
(notka na jutro :-) )
a no racja, ale position w metodzie, albo jako pole klasy .... Chyba jako pole klasy lepsze - logiczniejsze jak dla mnie.

Reply:
Lepiej w metodzie, bo tak to co zmianę position musiałbyś zmieniać ten obiekt, albo tworzyć nowy.
A to dalej ta sama bierka. To jak u człowieka. To że jest w Polsce czy w USA nie zmienia, że jest tym samym człowiekiem :) 
A possible moves zależą od position i board - to ładnie pokazuje funkcja (i patrząc na samą deklarację w interfejsie też jest to łatwe do odczytania). W przeciwnym przypadku zdaje się, jakby tylko board miał znaczenie. Takie funkcje i brak mutowania obiektu (zmieniania wartości position) ułatwia testowanie.